kaitaistruct.py

```diff
@@ -128,6 +128,9 @@ def is_eof(self):
         return self._io.tell() >= self.size()
 
     def seek(self, n):
+        if n < 0:
+            raise InvalidArgumentError("cannot seek to invalid position %d" % (n,))
```

Comment:
For the record, this exception was already thrown by the [`io.IOBase.seek()`](https://docs.python.org/3/library/io.html#io.IOBase.seek) method in the Python standard library. The exception type and message text and depended on which underlying stream was used. When using [`io.BytesIO`](https://docs.python.org/3/library/io.html#io.BytesIO), a `ValueError("negative seek value %zd")` was thrown (see [`Modules/_io/bytesio.c:659-660`](https://github.com/python/cpython/blob/401fff7423ca3c8bf1d02e594edfd1412616a559/Modules/_io/bytesio.c#L659-L660)). But when using a regular file, I'm getting `OSError: [Errno 22] Invalid argument` on both Windows and Linux.

So it seems better to do this check ourselves. Also a negative seek position can easily be a result of invalid input data (and thus the exception should be a subclass of `KaitaiStructError`) - imagine a situation like this:

```ksy
seq:
  - id: ofs_foo
    type: s4
instances:
  foo:
    pos: ofs_foo
    type: u1
```

If `ofs_foo` is negative and we attempt to read `foo`, it's a clear parse error.